### ISSUE TITLE
how to work on the single page website

### ISSUE THREAD

**Issue by wqh17101** [highlightjs/highlight.js#1900]
i have a single page website , and i use the highlight on one DOM subpage.
it works when i refresh the DOM subpage,but when i jump to this subpage from the index, it doesn`t work
## the index
![image](https://user-images.githubusercontent.com/26429138/47912105-714a6700-ded2-11e8-8c83-204eed9cd59e.png)
## click 
![image](https://user-images.githubusercontent.com/26429138/47912140-917a2600-ded2-11e8-823e-9ff25fa61344.png)
## refresh
![image](https://user-images.githubusercontent.com/26429138/47912183-b373a880-ded2-11e8-8182-1d27522301b9.png)

the difference i found  is that ,
![image](https://user-images.githubusercontent.com/26429138/47912246-e9189180-ded2-11e8-931b-661b56a70ae1.png)
![image](https://user-images.githubusercontent.com/26429138/47912262-f6358080-ded2-11e8-9c5a-038ebbbaa422.png)

**Comment by ouroboroscoding** [comment]
Yeah, I have the exact same problem. Calling hljs.initHighlighting() works once, but subsequent calls do nothing.

**Comment by ouroboroscoding** [comment]
@wqh17101 this is definitely hacky, but it worked for me.

`hljs.initHighlighting.called = false;`
`hljs.initHighlighting();`

I have a callback when the "page" changes, so I call this and it resets the highlighting.

**Comment by wqh17101** [comment]
i have found that highlightCode only run one time
so i hack it ,and run it where i need it
```
import hljs from 'highlight.js'
hljs.highlightCode =   function () { 
let blocks = document.querySelectorAll('pre  code');
[].forEach.call(blocks, hljs.highlightBlock);
 };
```

**Comment by yyyc514** [comment]
Watch out that `hljs.highlightBlock` is not intended to run over and over though if you run it on the same block... it will keep adding more and more tags each time... if you need to use it this way you should add a check inside hljs.highlightBlock to prevent it from highlighting block it's already highlighted in the past.

**Comment by yyyc514** [comment]
Ping.  Any lingering questions here?
